File: packages/registro/src/usecases/EditarRegistro.ts
import { CasoDeUso } from "common";
import { Registro } from "../model";
import { BackendRegistro } from "../provider";

type Entrada = { registro: Registro };

export class EditarRegistro implements CasoDeUso<Entrada, Response | void> {
  public constructor(private backend: BackendRegistro) {}
  async executar(dto: Entrada): Promise<Response | void> {
    return await this.backend.editar(dto.registro);
  }
}

File: apps/backend-springboot/src/main/java/br/com/dinheironapalmadamao/backend_springboot/core/registro/usecases/EditarRegistro.java
package br.com.dinheironapalmadamao.backend_springboot.core.registro.usecases;

import java.util.concurrent.CompletableFuture;

import br.com.dinheironapalmadamao.backend_springboot.core.common.CasoDeUso;
import br.com.dinheironapalmadamao.backend_springboot.core.registro.model.Registro;
import br.com.dinheironapalmadamao.backend_springboot.core.registro.provider.BackendRegistro;

public class EditarRegistro implements CasoDeUso<Registro, Void> {
  private BackendRegistro backend;

  public EditarRegistro(BackendRegistro backend) {
    this.backend = backend;
  }

  @Override
  public CompletableFuture<Void> excutar(Registro dto) {
    return CompletableFuture.runAsync(() -> {
      this.backend.editar(dto);
    });
  }
}
